nUpdate.Shared/Updating/UpdateResult.cs

```diff
@@ -48,46 +48,42 @@ internal class UpdateResult
                         continue;
 
 
-                    //Only check conditions if the remote conditions are not null or collection is empty
-                    if (config.Rolloutconditions != null && config.Rolloutconditions.Count != 0)
-                        //If the local conditions are null or the collection is empty then continue
+
+                    if (config.RolloutConditions != null && config.RolloutConditions.Count != 0)
                         if (conditions == null || !conditions.Any())
```

Comment:
I ask myself the following: Does it make sense to skip the updates when the user has no local conditions defined? In case that there are only positive conditions, this does make sense as no condition can be met (although it would be necessary). But, I am thinking about negative conditions. What if the developer just wants to exclude specific clients with the matching data (using negative conditions), but did not give every client a value for the used key (maybe cause he forgot it or it would not be necessary for him)? Then this client won't receive the update package although the intention was to only lock out users with an explicit negative value. Or would a hint such as "Note that every user must define a local value for this condition key once you define conditions for it. Otherwise, they won't receive the update." in the administration be clear enough?

Reply:
> Does it make sense to skip the updates when the user has no local conditions defined?

No.

If there are positive, negative and none conditions I would understand the usage as follows:
- None condition means the update is for **every** client.
- Positive conditions are used for a **exclusive** update for clients which met the conditions.
- Negative conditions are used to **exclude** clients based on a condition from an update.

I wouldn't allow positive and negative conditions for a single update because the meanings would conflict in this case.

Reply:
You are almost right but:
Positive and negative conditions do not necessarily result in conflict. If (example) a value with customernumber 1234 (positive) and a value with customernumber 3242 (negative) exists, there are no conflicts.

Reply:
The thing is that the conditons are currently interpreted as `or`, i. e. `Any` as you mentioned. With this specific case, we must implement both sides. In case of an `and`, one side would actually be enough.

Reply:
@FireEmerald please see on the case examples http://spsonline.remotewebaccess.com/sps/spssoftware/downloads/Conditions_examples.pdf

Reply:
@FireEmerald Of course, some checks will be addeed to prevent the user from adding a condition to both sides, resulting in a conflict.